src/ServiceContainer/SymfonyExtension.php

```diff
@@ -192,17 +192,21 @@ private function loadSharedKernelContainer(ContainerBuilder $container): void
 
     /**
      * @param ContainerBuilder $container
+     *
+     * @throws \Exception
      */
     private function loadKernelRebooter(ContainerBuilder $container): void
     {
-        $definition = new Definition(KernelRebooter::class, [new Reference(self::KERNEL_ID)]);
+        $definition = new Definition(KernelRebooter::class, [$container->get(self::KERNEL_ID)]);
```

Comment:
Does it really solve the issue?

Reply:
yes but I need to understand why!

Reply:
ok I found an explantation. I use in memory repository in my project. Like database, those repositories must be reset between scenarios. [Here](https://github.com/FriendsOfBehat/SymfonyExtension/blob/master/src/ServiceContainer/SymfonyExtension.php#L246) we give to `KernelBasedContainerAccessor` a "compiled or memory" container reference. It seems that need to give the same  "compiled or memory" container reference to the event subscriber.